packages/http/src/router/index.ts

```diff
@@ -63,36 +63,42 @@ export const router: IRouter<IHttpOperation, IHttpRequest, IHttpConfig> = {
       };
     });
 
+    matches = matches.filter(match => match.pathMatch !== MatchType.NOMATCH);
```

Comment:
this `route` function is so long and full of `if`s, mutations and throwing errors. The cognitive load one has to invest to understand it is a bit too much. Can this be refactored before it's too late?

Reply:
I want to ditch it completely and use a routing library, such as https://github.com/delvedor/find-my-way

Happy to see this refactored though.

Reply:
@XVincentX to your point on the library I'm happy to replace the code if you find a library that does what we need. We have a very special use case here - we not only route but check the server validity.

@lag-of-death I'm happy to see what I can do if you can suggest a refactoring.

Reply:
at least we could move all these `if`s that throw errors and make an array of "tuples" like so:
`[{predicate: () => false, action: () => throw 'A'}, {predicate: () => true, action: () => throw 'B'}]`
and move this array out of `route`

Reply:
Hey I'd like to leave that for some other time. I'm usually all for boy scouting but lots of things will change in the code base in next weeks/months so that may soon be irrelevant.